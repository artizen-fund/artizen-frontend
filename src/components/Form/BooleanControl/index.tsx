import { withJsonFormsControlProps } from '@jsonforms/react'
import type { Labels, JsonSchema, UISchemaElement } from '@jsonforms/core'
import { rankWith, schemaMatches } from '@jsonforms/core'
import SwitchControl from './SwitchControl'
import CheckboxControl from './CheckboxControl'

export interface BooleanControlProps {
  icon?: string
  invalid?: boolean
  label: string | Labels
  disabled?: boolean
  onChange?: (e: any) => void
  required?: boolean

  schema: JsonSchema
  uischema: UISchemaElement
  data: any
  handleChange(path: string, value: any): void
  path: string
}

export const BooleanControl = (props: BooleanControlProps) =>
  props.uischema.options?.format === 'switch' ? <SwitchControl {...props} /> : <CheckboxControl {...props} />

export const booleanControlTester = rankWith(
  3,
  schemaMatches(schema => schema.type === 'boolean'),
)

export default withJsonFormsControlProps(BooleanControl)
