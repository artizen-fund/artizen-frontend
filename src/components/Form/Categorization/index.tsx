// exploring: https://github.com/eclipsesource/jsonforms/tree/master/packages/vanilla/src/complex/categorization

import { JsonFormsDispatch } from '@jsonforms/react'
import { rankWith, uiTypeIs, and } from '@jsonforms/core'
import type { Categorization } from '@jsonforms/core'
import type { Labels, JsonSchema, UISchemaElement } from '@jsonforms/core'

export interface CategorizationProps {
  uischema: Categorization
  schema: JsonSchema
  path: string
}

const Categorization = ({ uischema, schema, path, ...props }: CategorizationProps) => {
  console.log('c', props, schema, uischema)
  return <div>c</div>
}

export const categorizationTester = rankWith(3, and(uiTypeIs('Categorization')))

export default Categorization
