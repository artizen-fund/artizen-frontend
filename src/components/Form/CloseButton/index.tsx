import styled from 'styled-components'
import { Icon } from '@components'
import { breakpoint } from '@theme'

const CloseButton = styled(props => <Icon {...props} />)<{ visible: boolean }>`
  position: absolute;
  z-index: 1002;
  top: 60px;
  right: 0px;
  @media only screen and (min-width: ${breakpoint.phablet}px) {
    top: 100px;
    right: 25px;
  }

  cursor: pointer;
  opacity: ${props => (props.visible ? 1 : 0)};
  pointer-events: ${props => (props.visible ? 'all' : 'none')};
`

export default CloseButton
