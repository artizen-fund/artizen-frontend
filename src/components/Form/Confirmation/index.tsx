import styled from 'styled-components'
import { rgba } from '@lib'
import { typography, palette } from '@theme'

const Confirmation = styled.div`
  display: none;
  grid-area: confirmation;
  flex-direction: column;
  justify-content: center;
  div {
    ${typography.title.l4}
    color: ${rgba(palette.moon)};
    margin-bottom: 0.25em;
  }
  p {
    ${typography.label.l1}
    color: ${rgba(palette.barracuda)};
  }
  text-align: center;
`
export default Confirmation
