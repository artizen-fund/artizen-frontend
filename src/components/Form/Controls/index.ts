export { default as StringControl, stringControlTester } from './StringControl'
export { default as NumberControl, numberControlTester } from './NumberControl'
export { default as BooleanControl, booleanControlTester } from './BooleanControl'
export { default as EnumControl, enumControlTester } from './EnumControl'
