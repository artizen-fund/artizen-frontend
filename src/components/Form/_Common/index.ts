export { default as InputIcon } from './InputIcon'
export { default as InputWrapper } from './InputWrapper'
export { default as Label } from './Label'
export { default as Message } from './Message'
export { default as Wrapper } from './Wrapper'
