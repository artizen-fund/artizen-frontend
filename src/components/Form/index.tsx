import { useState, cloneElement, useEffect } from 'react'
import { JsonForms } from '@jsonforms/react'
import { ErrorObject } from 'ajv'
import { JsonSchema, Layout, ControlElement } from '@jsonforms/core'
import { vanillaRenderers, vanillaCells } from '@jsonforms/vanilla-renderers'
import flattenChildren from 'react-keyed-flatten-children'
import { pickBy } from 'lodash'

import {
  StringControl,
  stringControlTester,
  NumberControl,
  numberControlTester,
  BooleanControl,
  booleanControlTester,
  EnumControl,
  enumControlTester,
} from './Controls'

interface FormProps<TStateInterface> {
  localStorageKey?: string
  schema: JsonSchema
  uischema: Layout
  data: TStateInterface
  setData: (input: TStateInterface) => void
  readonly?: boolean
  submitDisabledFromOutside?: boolean
  additionalErrors?: Array<ErrorObject>
  children?: React.ReactElement | Array<React.ReactElement>
}

type JsonFormsError = ErrorObject<string, Record<string, any>>

const Form = <TStateInterface extends Record<string, unknown>>({
  localStorageKey,
  schema,
  uischema,
  data,
  setData,
  readonly,
  submitDisabledFromOutside,
  additionalErrors,
  children,
  ...props
}: FormProps<TStateInterface>) => {
  console.log('data in here  ', data)
  const [disabled, setDisabled] = useState(true)
  const [formErrors, setFormErrors] = useState<Array<JsonFormsError>>()

  const setForm = (data: TStateInterface, errors?: Array<ErrorObject<string, Record<string, any>>>) => {
    setData(data)
    setFormErrors(errors)
  }

  useEffect(() => {
    if (!data) return

    if (localStorageKey && typeof localStorage !== 'undefined' && !!data) {
      const safeVars = (uischema.elements as Array<ControlElement>)
        .filter(schemaVar => !schemaVar.options?.unsafeToRetain)
        .map(schemaVar => schemaVar.scope.replace('#/properties/', ''))
      const safeData = pickBy(data, (_, key) => safeVars.includes(key))
      localStorage.setItem(localStorageKey, JSON.stringify(safeData))
    }

    setDisabled(
      formErrors === undefined ||
        [formErrors].flatMap(e => e).length > 0 ||
        (!!additionalErrors && [additionalErrors].flatMap(e => e).length > 0),
    )
  }, [data, formErrors, additionalErrors])

  const renderers = [
    ...vanillaRenderers,
    { tester: stringControlTester, renderer: StringControl },
    { tester: booleanControlTester, renderer: BooleanControl },
    { tester: numberControlTester, renderer: NumberControl },
    { tester: enumControlTester, renderer: EnumControl },
  ]

  return (
    <>
      <JsonForms
        {...{ schema, uischema, renderers, data, readonly, additionalErrors }}
        {...props}
        cells={vanillaCells}
        config={{ trim: true }}
        onChange={({ data, errors }) => setForm(data, errors)}
      />
      {flattenChildren(children).map(child =>
        cloneElement(child as React.ReactElement, { disabled: disabled || submitDisabledFromOutside }),
      )}
    </>
  )
}

export default Form
