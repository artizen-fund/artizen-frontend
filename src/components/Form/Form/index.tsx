import { useEffect, useState } from 'react'
import styled from 'styled-components'
import { JsonForms } from '@jsonforms/react'
import { vanillaRenderers, vanillaCells } from '@jsonforms/vanilla-renderers'
import { debounce } from 'lodash'
import StringControl, { stringControlTester } from '../StringControl'
import NumberControl, { numberControlTester } from '../NumberControl'
import BooleanControl, { booleanControlTester } from '../BooleanControl'
import EnumControl, { enumControlTester } from '../EnumControl'
import FormLabel, { formLabelTester } from '../FormLabel'
// import Categorization, { categorizationTester } from '../Categorization'
import { Button } from '@components'

interface FormProps {
  schema: any
  uischema: any
  initialState: any
}

const Form = ({ schema, uischema, initialState }: FormProps) => {
  const [data, setData] = useState<any>(undefined)
  const [errors, setErrors] = useState<any>(undefined)

  useEffect(() => {
    if (typeof localStorage === 'undefined') {
      setData(initialState)
      return
    }
    const frozenAnswers = localStorage.getItem(schema.name)
    if (!frozenAnswers) {
      setData(initialState)
      return
    }
    const thawedAnswers = JSON.parse(frozenAnswers)
    setData(thawedAnswers)
  }, [schema])

  const freezeAndSetData = debounce((newData: any, errors: any) => {
    setData(newData)
    setErrors(errors)
    if (typeof localStorage !== 'undefined') {
      localStorage.setItem(schema.name, JSON.stringify(newData))
    }
  }, 100)

  const renderers = [
    ...vanillaRenderers,
    { tester: stringControlTester, renderer: StringControl },
    { tester: booleanControlTester, renderer: BooleanControl },
    { tester: numberControlTester, renderer: NumberControl },
    { tester: enumControlTester, renderer: EnumControl },
    { tester: formLabelTester, renderer: FormLabel },
  ]

  return (
    <Wrapper>
      {data && (
        <JsonForms
          {...{ uischema, schema, renderers, data }}
          config={{
            trim: true,
          }}
          cells={vanillaCells}
          onChange={({ data, errors }) => freezeAndSetData(data, errors)}
        />
      )}
      <Button onClick={() => console.log('saving!', data)} disabled={errors?.length > 0}>
        Submit
      </Button>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  margin: 20px;
  .vertical-layout {
    display: flex;
    flex-direction: column;
  }
  .vertical-layout-item {
  }
  .horizontal-layout {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    gap: 0 50px;
  }
  .horizontal-layout-item {
    flex: 1;
  }
`

export default Form
