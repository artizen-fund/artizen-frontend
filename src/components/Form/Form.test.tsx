import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Form from './'
import { schema, uischema, initialState } from '@forms/sample'

describe('Form', () => {
  it('renders Form unchanged', () => {
    const { container } = render(<Form {...{ schema, uischema, initialState }} />)
    expect(container).toMatchSnapshot()
  })
})

/* 
Possible future testsâ€¦

- onchange triggers console log
- error state is respected

Note: @forms/sample is a beefy form.
			If this test lags, maybe consider feeding a simpler form.
*/
