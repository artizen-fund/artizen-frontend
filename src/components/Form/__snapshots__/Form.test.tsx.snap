// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders Form unchanged 1`] = `
<div>
  <div
    class="vertical-layout"
  >
    <div
      class="vertical-layout-item"
    >
      <label
        class=""
      >
        Section Label
      </label>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-2f9d5146-0 duTmMz"
        id="#/properties/stringExample"
      >
        <div
          class="sc-233fee17-0 kLNnFl"
        >
          <input
            maxlength="20"
            minlength="3"
            placeholder=" "
            required=""
            type="text"
            value=""
          />
          <label
            class="sc-ffdb2e6c-0 eexGRl"
          >
            3-20 Character Length String
             *
          </label>
        </div>
        <div
          class="sc-955ed11e-0 griBRj"
        >
          * required field
        </div>
      </div>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-2f9d5146-0 kskevu"
        id="#/properties/phoneExample"
      >
        <div
          class="sc-233fee17-0 kLNnFl"
        >
          <div
            class="sc-f5b34ffa-0 iLecjc PhoneInput"
          >
            <div
              class="PhoneInputCountry"
            >
              <select
                aria-label="Phone number country"
                class="PhoneInputCountrySelect"
              >
                <option
                  value="ZZ"
                >
                  International
                </option>
                <option
                  value="AF"
                >
                  Afghanistan
                </option>
                <option
                  value="AX"
                >
                  Åland Islands
                </option>
                <option
                  value="AL"
                >
                  Albania
                </option>
                <option
                  value="DZ"
                >
                  Algeria
                </option>
                <option
                  value="AS"
                >
                  American Samoa
                </option>
                <option
                  value="AD"
                >
                  Andorra
                </option>
                <option
                  value="AO"
                >
                  Angola
                </option>
                <option
                  value="AI"
                >
                  Anguilla
                </option>
                <option
                  value="AG"
                >
                  Antigua and Barbuda
                </option>
                <option
                  value="AR"
                >
                  Argentina
                </option>
                <option
                  value="AM"
                >
                  Armenia
                </option>
                <option
                  value="AW"
                >
                  Aruba
                </option>
                <option
                  value="AC"
                >
                  Ascension Island
                </option>
                <option
                  value="AU"
                >
                  Australia
                </option>
                <option
                  value="AT"
                >
                  Austria
                </option>
                <option
                  value="AZ"
                >
                  Azerbaijan
                </option>
                <option
                  value="BS"
                >
                  Bahamas
                </option>
                <option
                  value="BH"
                >
                  Bahrain
                </option>
                <option
                  value="BD"
                >
                  Bangladesh
                </option>
                <option
                  value="BB"
                >
                  Barbados
                </option>
                <option
                  value="BY"
                >
                  Belarus
                </option>
                <option
                  value="BE"
                >
                  Belgium
                </option>
                <option
                  value="BZ"
                >
                  Belize
                </option>
                <option
                  value="BJ"
                >
                  Benin
                </option>
                <option
                  value="BM"
                >
                  Bermuda
                </option>
                <option
                  value="BT"
                >
                  Bhutan
                </option>
                <option
                  value="BO"
                >
                  Bolivia
                </option>
                <option
                  value="BQ"
                >
                  Bonaire, Sint Eustatius and Saba
                </option>
                <option
                  value="BA"
                >
                  Bosnia and Herzegovina
                </option>
                <option
                  value="BW"
                >
                  Botswana
                </option>
                <option
                  value="BR"
                >
                  Brazil
                </option>
                <option
                  value="IO"
                >
                  British Indian Ocean Territory
                </option>
                <option
                  value="BN"
                >
                  Brunei Darussalam
                </option>
                <option
                  value="BG"
                >
                  Bulgaria
                </option>
                <option
                  value="BF"
                >
                  Burkina Faso
                </option>
                <option
                  value="MM"
                >
                  Burma
                </option>
                <option
                  value="BI"
                >
                  Burundi
                </option>
                <option
                  value="KH"
                >
                  Cambodia
                </option>
                <option
                  value="CM"
                >
                  Cameroon
                </option>
                <option
                  value="CA"
                >
                  Canada
                </option>
                <option
                  value="CV"
                >
                  Cape Verde
                </option>
                <option
                  value="KY"
                >
                  Cayman Islands
                </option>
                <option
                  value="CF"
                >
                  Central African Republic
                </option>
                <option
                  value="TD"
                >
                  Chad
                </option>
                <option
                  value="CL"
                >
                  Chile
                </option>
                <option
                  value="CN"
                >
                  China
                </option>
                <option
                  value="CX"
                >
                  Christmas Island
                </option>
                <option
                  value="CC"
                >
                  Cocos (Keeling) Islands
                </option>
                <option
                  value="CO"
                >
                  Colombia
                </option>
                <option
                  value="KM"
                >
                  Comoros
                </option>
                <option
                  value="CG"
                >
                  Congo
                </option>
                <option
                  value="CD"
                >
                  Congo, Democratic Republic of the
                </option>
                <option
                  value="CK"
                >
                  Cook Islands
                </option>
                <option
                  value="CR"
                >
                  Costa Rica
                </option>
                <option
                  value="CI"
                >
                  Cote d'Ivoire
                </option>
                <option
                  value="HR"
                >
                  Croatia
                </option>
                <option
                  value="CU"
                >
                  Cuba
                </option>
                <option
                  value="CW"
                >
                  Curaçao
                </option>
                <option
                  value="CY"
                >
                  Cyprus
                </option>
                <option
                  value="CZ"
                >
                  Czech Republic
                </option>
                <option
                  value="DK"
                >
                  Denmark
                </option>
                <option
                  value="DJ"
                >
                  Djibouti
                </option>
                <option
                  value="DM"
                >
                  Dominica
                </option>
                <option
                  value="DO"
                >
                  Dominican Republic
                </option>
                <option
                  value="EC"
                >
                  Ecuador
                </option>
                <option
                  value="EG"
                >
                  Egypt
                </option>
                <option
                  value="SV"
                >
                  El Salvador
                </option>
                <option
                  value="GQ"
                >
                  Equatorial Guinea
                </option>
                <option
                  value="ER"
                >
                  Eritrea
                </option>
                <option
                  value="EE"
                >
                  Estonia
                </option>
                <option
                  value="ET"
                >
                  Ethiopia
                </option>
                <option
                  value="FK"
                >
                  Falkland Islands
                </option>
                <option
                  value="FO"
                >
                  Faroe Islands
                </option>
                <option
                  value="FM"
                >
                  Federated States of Micronesia
                </option>
                <option
                  value="FJ"
                >
                  Fiji
                </option>
                <option
                  value="FI"
                >
                  Finland
                </option>
                <option
                  value="FR"
                >
                  France
                </option>
                <option
                  value="GF"
                >
                  French Guiana
                </option>
                <option
                  value="PF"
                >
                  French Polynesia
                </option>
                <option
                  value="GA"
                >
                  Gabon
                </option>
                <option
                  value="GM"
                >
                  Gambia
                </option>
                <option
                  value="GE"
                >
                  Georgia
                </option>
                <option
                  value="DE"
                >
                  Germany
                </option>
                <option
                  value="GH"
                >
                  Ghana
                </option>
                <option
                  value="GI"
                >
                  Gibraltar
                </option>
                <option
                  value="GR"
                >
                  Greece
                </option>
                <option
                  value="GL"
                >
                  Greenland
                </option>
                <option
                  value="GD"
                >
                  Grenada
                </option>
                <option
                  value="GP"
                >
                  Guadeloupe
                </option>
                <option
                  value="GU"
                >
                  Guam
                </option>
                <option
                  value="GT"
                >
                  Guatemala
                </option>
                <option
                  value="GG"
                >
                  Guernsey
                </option>
                <option
                  value="GN"
                >
                  Guinea
                </option>
                <option
                  value="GW"
                >
                  Guinea-Bissau
                </option>
                <option
                  value="GY"
                >
                  Guyana
                </option>
                <option
                  value="HT"
                >
                  Haiti
                </option>
                <option
                  value="VA"
                >
                  Holy See (Vatican City State)
                </option>
                <option
                  value="HN"
                >
                  Honduras
                </option>
                <option
                  value="HK"
                >
                  Hong Kong
                </option>
                <option
                  value="HU"
                >
                  Hungary
                </option>
                <option
                  value="IS"
                >
                  Iceland
                </option>
                <option
                  value="IN"
                >
                  India
                </option>
                <option
                  value="ID"
                >
                  Indonesia
                </option>
                <option
                  value="IR"
                >
                  Iran
                </option>
                <option
                  value="IQ"
                >
                  Iraq
                </option>
                <option
                  value="IE"
                >
                  Ireland
                </option>
                <option
                  value="IM"
                >
                  Isle of Man
                </option>
                <option
                  value="IL"
                >
                  Israel
                </option>
                <option
                  value="IT"
                >
                  Italy
                </option>
                <option
                  value="JM"
                >
                  Jamaica
                </option>
                <option
                  value="JP"
                >
                  Japan
                </option>
                <option
                  value="JE"
                >
                  Jersey
                </option>
                <option
                  value="JO"
                >
                  Jordan
                </option>
                <option
                  value="KZ"
                >
                  Kazakhstan
                </option>
                <option
                  value="KE"
                >
                  Kenya
                </option>
                <option
                  value="KI"
                >
                  Kiribati
                </option>
                <option
                  value="XK"
                >
                  Kosovo
                </option>
                <option
                  value="KW"
                >
                  Kuwait
                </option>
                <option
                  value="KG"
                >
                  Kyrgyzstan
                </option>
                <option
                  value="LA"
                >
                  Laos
                </option>
                <option
                  value="LV"
                >
                  Latvia
                </option>
                <option
                  value="LB"
                >
                  Lebanon
                </option>
                <option
                  value="LS"
                >
                  Lesotho
                </option>
                <option
                  value="LR"
                >
                  Liberia
                </option>
                <option
                  value="LY"
                >
                  Libya
                </option>
                <option
                  value="LI"
                >
                  Liechtenstein
                </option>
                <option
                  value="LT"
                >
                  Lithuania
                </option>
                <option
                  value="LU"
                >
                  Luxembourg
                </option>
                <option
                  value="MO"
                >
                  Macao
                </option>
                <option
                  value="MG"
                >
                  Madagascar
                </option>
                <option
                  value="MW"
                >
                  Malawi
                </option>
                <option
                  value="MY"
                >
                  Malaysia
                </option>
                <option
                  value="MV"
                >
                  Maldives
                </option>
                <option
                  value="ML"
                >
                  Mali
                </option>
                <option
                  value="MT"
                >
                  Malta
                </option>
                <option
                  value="MH"
                >
                  Marshall Islands
                </option>
                <option
                  value="MQ"
                >
                  Martinique
                </option>
                <option
                  value="MR"
                >
                  Mauritania
                </option>
                <option
                  value="MU"
                >
                  Mauritius
                </option>
                <option
                  value="YT"
                >
                  Mayotte
                </option>
                <option
                  value="MX"
                >
                  Mexico
                </option>
                <option
                  value="MD"
                >
                  Moldova
                </option>
                <option
                  value="MC"
                >
                  Monaco
                </option>
                <option
                  value="MN"
                >
                  Mongolia
                </option>
                <option
                  value="ME"
                >
                  Montenegro
                </option>
                <option
                  value="MS"
                >
                  Montserrat
                </option>
                <option
                  value="MA"
                >
                  Morocco
                </option>
                <option
                  value="MZ"
                >
                  Mozambique
                </option>
                <option
                  value="NA"
                >
                  Namibia
                </option>
                <option
                  value="NR"
                >
                  Nauru
                </option>
                <option
                  value="NP"
                >
                  Nepal
                </option>
                <option
                  value="NL"
                >
                  Netherlands
                </option>
                <option
                  value="NC"
                >
                  New Caledonia
                </option>
                <option
                  value="NZ"
                >
                  New Zealand
                </option>
                <option
                  value="NI"
                >
                  Nicaragua
                </option>
                <option
                  value="NE"
                >
                  Niger
                </option>
                <option
                  value="NG"
                >
                  Nigeria
                </option>
                <option
                  value="NU"
                >
                  Niue
                </option>
                <option
                  value="NF"
                >
                  Norfolk Island
                </option>
                <option
                  value="KP"
                >
                  North Korea
                </option>
                <option
                  value="MK"
                >
                  North Macedonia
                </option>
                <option
                  value="MP"
                >
                  Northern Mariana Islands
                </option>
                <option
                  value="NO"
                >
                  Norway
                </option>
                <option
                  value="OM"
                >
                  Oman
                </option>
                <option
                  value="PK"
                >
                  Pakistan
                </option>
                <option
                  value="PW"
                >
                  Palau
                </option>
                <option
                  value="PS"
                >
                  Palestine
                </option>
                <option
                  value="PA"
                >
                  Panama
                </option>
                <option
                  value="PG"
                >
                  Papua New Guinea
                </option>
                <option
                  value="PY"
                >
                  Paraguay
                </option>
                <option
                  value="PE"
                >
                  Peru
                </option>
                <option
                  value="PH"
                >
                  Philippines
                </option>
                <option
                  value="PL"
                >
                  Poland
                </option>
                <option
                  value="PT"
                >
                  Portugal
                </option>
                <option
                  value="PR"
                >
                  Puerto Rico
                </option>
                <option
                  value="QA"
                >
                  Qatar
                </option>
                <option
                  value="RE"
                >
                  Reunion
                </option>
                <option
                  value="RO"
                >
                  Romania
                </option>
                <option
                  value="RU"
                >
                  Russia
                </option>
                <option
                  value="RW"
                >
                  Rwanda
                </option>
                <option
                  value="BL"
                >
                  Saint Barthélemy
                </option>
                <option
                  value="SH"
                >
                  Saint Helena
                </option>
                <option
                  value="KN"
                >
                  Saint Kitts and Nevis
                </option>
                <option
                  value="LC"
                >
                  Saint Lucia
                </option>
                <option
                  value="MF"
                >
                  Saint Martin (French Part)
                </option>
                <option
                  value="PM"
                >
                  Saint Pierre and Miquelon
                </option>
                <option
                  value="VC"
                >
                  Saint Vincent and the Grenadines
                </option>
                <option
                  value="WS"
                >
                  Samoa
                </option>
                <option
                  value="SM"
                >
                  San Marino
                </option>
                <option
                  value="ST"
                >
                  Sao Tome and Principe
                </option>
                <option
                  value="SA"
                >
                  Saudi Arabia
                </option>
                <option
                  value="SN"
                >
                  Senegal
                </option>
                <option
                  value="RS"
                >
                  Serbia
                </option>
                <option
                  value="SC"
                >
                  Seychelles
                </option>
                <option
                  value="SL"
                >
                  Sierra Leone
                </option>
                <option
                  value="SG"
                >
                  Singapore
                </option>
                <option
                  value="SX"
                >
                  Sint Maarten
                </option>
                <option
                  value="SK"
                >
                  Slovakia
                </option>
                <option
                  value="SI"
                >
                  Slovenia
                </option>
                <option
                  value="SB"
                >
                  Solomon Islands
                </option>
                <option
                  value="SO"
                >
                  Somalia
                </option>
                <option
                  value="ZA"
                >
                  South Africa
                </option>
                <option
                  value="KR"
                >
                  South Korea
                </option>
                <option
                  value="SS"
                >
                  South Sudan
                </option>
                <option
                  value="ES"
                >
                  Spain
                </option>
                <option
                  value="LK"
                >
                  Sri Lanka
                </option>
                <option
                  value="SD"
                >
                  Sudan
                </option>
                <option
                  value="SR"
                >
                  Suriname
                </option>
                <option
                  value="SJ"
                >
                  Svalbard and Jan Mayen
                </option>
                <option
                  value="SZ"
                >
                  Swaziland
                </option>
                <option
                  value="SE"
                >
                  Sweden
                </option>
                <option
                  value="CH"
                >
                  Switzerland
                </option>
                <option
                  value="SY"
                >
                  Syria
                </option>
                <option
                  value="TW"
                >
                  Taiwan
                </option>
                <option
                  value="TJ"
                >
                  Tajikistan
                </option>
                <option
                  value="TZ"
                >
                  Tanzania
                </option>
                <option
                  value="TH"
                >
                  Thailand
                </option>
                <option
                  value="TL"
                >
                  Timor-Leste
                </option>
                <option
                  value="TG"
                >
                  Togo
                </option>
                <option
                  value="TK"
                >
                  Tokelau
                </option>
                <option
                  value="TO"
                >
                  Tonga
                </option>
                <option
                  value="TT"
                >
                  Trinidad and Tobago
                </option>
                <option
                  value="TA"
                >
                  Tristan da Cunha
                </option>
                <option
                  value="TN"
                >
                  Tunisia
                </option>
                <option
                  value="TR"
                >
                  Turkey
                </option>
                <option
                  value="TM"
                >
                  Turkmenistan
                </option>
                <option
                  value="TC"
                >
                  Turks and Caicos Islands
                </option>
                <option
                  value="TV"
                >
                  Tuvalu
                </option>
                <option
                  value="UG"
                >
                  Uganda
                </option>
                <option
                  value="UA"
                >
                  Ukraine
                </option>
                <option
                  value="AE"
                >
                  United Arab Emirates
                </option>
                <option
                  value="GB"
                >
                  United Kingdom
                </option>
                <option
                  value="US"
                >
                  United States
                </option>
                <option
                  value="UY"
                >
                  Uruguay
                </option>
                <option
                  value="UZ"
                >
                  Uzbekistan
                </option>
                <option
                  value="VU"
                >
                  Vanuatu
                </option>
                <option
                  value="VE"
                >
                  Venezuela
                </option>
                <option
                  value="VN"
                >
                  Vietnam
                </option>
                <option
                  value="VG"
                >
                  Virgin Islands, British
                </option>
                <option
                  value="VI"
                >
                  Virgin Islands, U.S.
                </option>
                <option
                  value="WF"
                >
                  Wallis and Futuna
                </option>
                <option
                  value="EH"
                >
                  Western Sahara
                </option>
                <option
                  value="YE"
                >
                  Yemen
                </option>
                <option
                  value="ZM"
                >
                  Zambia
                </option>
                <option
                  value="ZW"
                >
                  Zimbabwe
                </option>
              </select>
              <div
                class="PhoneInputCountryIconUnicode"
              >
                🇺🇸
              </div>
              <div
                class="PhoneInputCountrySelectArrow"
              />
            </div>
            <input
              autocomplete="tel"
              class="PhoneInputInput"
              placeholder=" "
              type="tel"
              value=""
            />
          </div>
          <label
            class="sc-ffdb2e6c-0 bixQNG"
          >
            Phone Example
          </label>
        </div>
        <div
          class="sc-955ed11e-0 griBRj"
        />
      </div>
    </div>
    <div
      class="vertical-layout-item"
    >
      <label
        class="sc-accf1c1-0 RJQBw"
        id="#/properties/switchBooleanExample"
      >
        <span
          class="sc-accf1c1-3 bLaMtz"
        >
          Switch Boolean Example
        </span>
        <input
          class="sc-accf1c1-1 fxPrqN"
          type="checkbox"
        />
        <span
          class="sc-accf1c1-2 jDOa-Dy"
        />
      </label>
    </div>
    <div
      class="vertical-layout-item"
    >
      <label
        class="sc-32dd2a74-4 hluhiq"
        id="#/properties/booleanCheckExample"
      >
        <div
          class="sc-32dd2a74-0 jdmmOy"
        >
          <input
            class="sc-32dd2a74-1 iRKuyc"
            type="checkbox"
          />
          <span
            class="sc-32dd2a74-2 hvqdrD"
          />
        </div>
        <span
          class="sc-32dd2a74-3 jLQRRt"
        >
          Boolean Check Example
        </span>
      </label>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-2f9d5146-0 fQETFB"
        id="#/properties/enumExample"
      >
        <div
          class="sc-233fee17-0 kLNnFl"
        >
          <select
            class="noData"
            placeholder=" "
          >
            <option
              value=""
            />
            <option
              value="DE"
            >
              DE
            </option>
            <option
              value="IT"
            >
              IT
            </option>
            <option
              value="JP"
            >
              JP
            </option>
            <option
              value="US"
            >
              US
            </option>
            <option
              value="RU"
            >
              RU
            </option>
            <option
              value="Other"
            >
              Other
            </option>
          </select>
          <div
            class="sc-b0af9a1b-0 dENUF"
          />
          <label
            class="sc-ffdb2e6c-0 eexGRl noData"
          >
            Enum Example
          </label>
        </div>
        <div
          class="sc-955ed11e-0 griBRj"
        />
      </div>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-1490f8e8-0 hDZEyp"
        id="#/properties/enumSegmentsExample"
        label="Enum Segments Example"
      >
        <label
          class="sc-1490f8e8-1 vGKrz"
        >
          <input
            type="radio"
            value="vanilla"
          />
          <span>
            vanilla
          </span>
        </label>
        <label
          class="sc-1490f8e8-1 vGKrz"
        >
          <input
            type="radio"
            value="chocolate"
          />
          <span>
            chocolate
          </span>
        </label>
        <label
          class="sc-1490f8e8-1 vGKrz"
        >
          <input
            type="radio"
            value="strawberry"
          />
          <span>
            strawberry
          </span>
        </label>
        <label
          class="sc-1490f8e8-1 vGKrz"
        >
          <input
            type="radio"
            value="neapolitan"
          />
          <span>
            neapolitan
          </span>
        </label>
      </div>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-2f9d5146-0 HhwJg"
        id="#/properties/integerExample"
      >
        <div
          class="sc-233fee17-0 kLNnFl"
        >
          <input
            class="noData"
            maxlength="100"
            minlength="1"
            placeholder=" "
            step="1"
            type="number"
            value=""
          />
          <label
            class="sc-ffdb2e6c-0 eexGRl"
          >
            Integer Example
          </label>
        </div>
        <div
          class="sc-955ed11e-0 griBRj"
        />
      </div>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-2f9d5146-0 dTiGtl"
        id="#/properties/numberExample"
      >
        <div
          class="sc-233fee17-0 kLNnFl"
        >
          <input
            class="noData"
            placeholder=" "
            step="any"
            type="number"
            value=""
          />
          <label
            class="sc-ffdb2e6c-0 eexGRl"
          >
            Number Example
          </label>
        </div>
        <div
          class="sc-955ed11e-0 griBRj"
        />
      </div>
    </div>
    <div
      class="vertical-layout-item"
    >
      <div
        class="sc-2f9d5146-0 liLUtm"
        id="#/properties/emailExample"
      >
        <div
          class="sc-233fee17-0 kLNnFl"
        >
          <input
            placeholder="your@email.com"
            type="text"
            value=""
          />
          <label
            class="sc-ffdb2e6c-0 eexGRl"
          >
            Email Example
          </label>
        </div>
        <div
          class="sc-955ed11e-0 griBRj"
        />
      </div>
    </div>
  </div>
</div>
`;
